cmd/podman/commit.go

```diff
@@ -77,35 +77,50 @@ func commitCmd(c *cli.Context) error {
 		return errors.Errorf("too many arguments. Usage CONTAINER [REFERENCE]")
 	}
 
-	changes := v1.ImageConfig{}
-	if c.IsSet("change") {
-		changes, err = getImageConfig(c.StringSlice("change"))
-		if err != nil {
-			return errors.Wrapf(err, "error adding config changes to image %q", container)
-		}
-	}
+	//changes := v1.ImageConfig{}
+	//if c.IsSet("change") {
+	//	changes, err = getImageConfig(c.StringSlice("change"))
+	//	if err != nil {
+	//		return errors.Wrapf(err, "error adding config changes to image %q", container)
+	//	}
+	//}
 
-	history := []v1.History{
-		{Comment: c.String("message")},
-	}
+	//history := []v1.History{
+	//	{Comment: c.String("message")},
+	//}
 
-	config := v1.Image{
-		Config:  changes,
-		History: history,
-		Author:  c.String("author"),
+	//config := v1.Image{
+	//	Config:  changes,
+	//	History: history,
+	//	Author:  c.String("author"),
+	//}
+
+	if c.IsSet("change") {
+		for _, change := range c.StringSlice("change") {
+			splitChange := strings.Split(strings.ToUpper(change), "=")
+			if !util.StringInSlice(splitChange[0], []string{"CMD", "ENTRYPOINT", "ENV", "EXPOSE", "LABEL", "STOPSIGNAL", "USER", "VOLUME", "WORKDIR"}) {
+				return errors.Errorf("invalid syntax for --change ", change)
+			}
+		}
 	}
 
 	if !c.Bool("quiet") {
 		writer = os.Stderr
 	}
-
 	ctr, err := runtime.LookupContainer(container)
 	if err != nil {
 		return errors.Wrapf(err, "error looking up container %q", container)
 	}
-	newImage, err := ctr.Commit(c.BoolT("pause"), reference, writer, image.SigningOptions{}, config)
-	if err == nil {
-		fmt.Println(newImage.ID())
+
+	newImage := runtime.ImageRuntime().NewNoExist(reference)
```

Comment:
Separate from the Buildah stuff, but I really don't like this... can we just pass in a string for the name of the new image?

Reply:
We can give return a *image.Image from commit so we still get a good image to work with